app/src/main/java/com/example/weatherapp/MainActivity.kt

```diff
@@ -0,0 +1,69 @@
+package com.example.weatherapp
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.lifecycle.Observer
+import androidx.work.*
+import com.example.weatherapp.databinding.ActivityMainBinding
+import com.example.weatherapp.workers.RequestWeatherWorker
+
+class MainActivity : AppCompatActivity() {
+    private var lat = 53.43
+    private var lon = 14.55
+    private var API_KEY = "680f2f4fd0536de70ad6a0f3bcd81790"
+    private var binding: ActivityMainBinding? = null
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding?.root)
+
+        binding?.buttonGetWeather?.setOnClickListener {
+            buttonGetWeatherListener()
+        }
+    }
+
+    private fun buttonGetWeatherListener() {
```

Comment:
I suggest refactoring name of this function as there's no implementation of any actual Listener in the body.

Reply:
I renamed it to enqueueRequestWeatherWorker but I'll think of a more appropriate name